ansible/library/generate_golden_config_db.py

```diff
@@ -211,15 +218,21 @@ def generate_smartswitch_golden_config_db(self):
         if "DHCP_SERVER_IPV4_PORT" not in ori_config_db:
             ori_config_db["DHCP_SERVER_IPV4_PORT"] = {}
 
+        hwsku_config = smartswitch_hwsku_config[hwsku]
         for i in range(smartswitch_hwsku_config[hwsku]["dpu_num"]):
-            port_key = smartswitch_hwsku_config[hwsku]["port_key"].format(i)
-            interface_key = smartswitch_hwsku_config[hwsku]["interface_key"].format(i, i)
-            dpu_key = smartswitch_hwsku_config[hwsku]["dpu_key"].format(i)
+            if "base" in hwsku_config and "step" in hwsku_config:
+                port_key = hwsku_config["port_key"].format(hwsku_config["base"] + i * hwsku_config["step"])
+            else:
+                port_key = hwsku_config["port_key"].format(i)
+            if "interface_key" in hwsku_config:
+                interface_key = hwsku_config["interface_key"].format(i, i)
+            dpu_key = hwsku_config["dpu_key"].format(i)
 
             if port_key in ori_config_db["PORT"]:
                 ori_config_db["PORT"][port_key]["admin_status"] = "up"
-                ori_config_db["INTERFACE"][port_key] = {}
-                ori_config_db["INTERFACE"][interface_key] = {}
+                if "interface_key" in hwsku_config:
+                    ori_config_db["INTERFACE"][port_key] = {}
```

Comment:
Should this line be before the if "interface_key" check?

Reply:
From the configuration in starlab, it should not be taken before.
Once there is ip prefix, it will add port_key to INTERFACE table.